src/main/java/com/puppycrawl/tools/checkstyle/api/Violation.java

```diff
@@ -369,7 +369,7 @@ public static void setLocale(Locale locale) {
             sLocale = Locale.ROOT;
         }
         else {
-            sLocale = locale;
+            sLocale = (Locale) locale.clone();
```

Comment:
This one is a true-positve. Fixed.

Reply:
This also turned out to be false-positive. The class `Local` is immutable class. Reverted.